[Lim Shao Cong] iP

arcornior opened the pull request:
# ENCIK
### GOOD MORNING GENNERMEN!
CANNOT REMEMBER WHAT TASK YOU NEED TO DO ISIT? NEBBERMIND, LET ENCIK TAKE CARE OF IT FOR YOU! ENCIK IS
* TEXT ACTIVATED
* SIBEI HELPFUL
* SIBEI SCARY
* ~~GOING TO MAKE YOU SIGN EXTRA~~
* ENCIK! NOT OFFICER! BLOODY HELL GET YOUR ACT TOGETHER RECRUIT! 🤬 

## ALL YOU NEED TO DO IS:
1. DOWNLOAD ENCIK FROM [HERE!](https://github.com/nus-cs2103-AY2122S2/ip)
2. RUN ENCIK!
3. ADD YOUR TASK!
4. LET ENCIK HELP YOU!
5. ~~SIGN 2 EXTRA!~~

AND ENCIK IS ~~WORTH 2 EXTRAS!~~ **FREE!**

## ENCIK FEATURES:
- [x] MANAGING YOUR GUARD DUTIES!
- [x] REMINDERS TO DOWN 20!
- [ ] HOUSE VISIT WHEN YOU REPORT SICK! 
---
IF YOU ARE JAVA TRAINED, ENCIK CAN ALSO HELP YOU PRACTICE JAVA! YOU COME SEE ENCIK'S `main` METHOD! ENCIK NOT BAD RIGHT!
> "MY AH MA CAN CODE BETTER THAN YOU CHAO RECRUIT!" - ENCIK
```
    public static void main(String[] args) {
        new Duke(LOG_PATH).run();
    }
```

Commit:
Add Gradle support

Commit:
Edited Duke.java
Duke, now known as Encik, will echo user input until bye is input

Commit:
Added function format to format output
Added borders to echoed tasks

Commit:
Duke.java - Fixed formatting, can now print tasks entered by user
Added Task.java to keep track of user's tasks

Commit:
Duke.java
- Added ability to mark/ unmark tasks
Task.java
- Added boolean to keep track of task's completion

Commit:
Duke.java
- Added tracking for 3 types of tasks
Task.java
- Supports tracking of 3 types of tasks

Commit:
Duke.java
- Formatted output

Commit:
Duke.java
- Edited to support inheritance
Task.java
- Edited to support inheritance
Added Deadline.java
Added Event.java
Added ToDo.java

Commit:
Fixed bug where ToDo was printing wrong output

Commit:
Added input and EXPECTED files for automated testing

Commit:
Added testing for list function

Commit:
Replaced hardcoded spaces with final values instead

Commit:
Added DukeException class for handling exceptions

Commit:
Added error handling for incomplete description, wrong command

Commit:
Replaced hardcoded indent with SPACE

Commit:
Added delete command
Fixed unmark error message, formatting

Commit:
Fixed error in Duke with string
Updataed input and Expected files

Commit:
Fix minor indent issue

Commit:
Add support for logging of previous tasks

Previous tasks saved in Duke now carry over to a new instance
of Duke.

Commit:
Add support for storing dates/times as LocalDateTime object

Commit:
Revert "Add support for logging of previous tasks"

This reverts commit 24060d8e7e232e3e35157074b97ea0ed02d984a9.

Commit:
Merge branch 'branch-Level-8'

Commit:
Fix bugs relating to logging

Commit:
Fix Events not keeping track of Event end time.

RyanCheungJF commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,188 @@
+                    String cmd = temp[0];
+                    //placeholder in case temp[1] is empty and cannot be initialised
+                    String rem = "";
+                    boolean incomplete = false;
```
I believe the boolean name needs to be changed according to the coding standard.

![image](https://user-images.githubusercontent.com/63341400/151937900-9bdaca84-bbb3-49ee-ab97-6e0c8c5e4112.png)


```suggestion
                    boolean isIncomplete = false;
```

RyanCheungJF commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,188 @@
+                        } else {
+                            int num = Integer.parseInt(rem) - 1;
+                            tasks.get(num).mark();
+                            String confirm =
```
Not too sure if this violates code quality, but was wondering if the new line was necessary? (Line 78)

RyanCheungJF commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,188 @@
+
 public class Duke {
+    private static void format(String output) {
+        String border = "   ____________________________________________________________";
```
Can consider making this a constant

RyanCheungJF reviewed (commented):
Overall looks good! Missing javadocs but the rest of the naming conventions and code quality is adhered to!

Gernene commented on src/main/java/DukeFile.java:
```diff
@@ -0,0 +1,111 @@
+        log.close();
+    }
+
+    public ArrayList<Task> readTasks() {
```
Don't forget to add header comments. :3

Gernene commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,94 @@
+        }
+    }
+
+    public String getName(){
```
Missed a little space between () and {

Gernene commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,188 @@
+                        if (incomplete) {
+                            throw new DukeException("WHAT YOU WANT DO? NOTHING AH HELLOOOOOO?");
+                        } else {
+                            String task = rem.split(" /at ")[0];
```
Maybe you could consider making the "rem" variable name more descriptive?

Gernene commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,188 @@
+                            format(confirm);
+                        }
+                    } else if (cmd.equals("delete")) {
+                        if (incomplete) {
```
Maybe consider "isIncomplete" as an alternative to "incomplete"?

Gernene commented on src/main/java/DukeFile.java:
```diff
@@ -0,0 +1,111 @@
+        File d = new File(dir);
+        File f = new File(path);
+        if (!d.exists()) {
+            d.mkdirs();
```
I think "mkdirs" should be "mkDirs" if you're following the standard. I'd also double-check abbreviations vs standard English.

Gernene reviewed (commented):
Hello! I made a few comments on a couple of things you might want to double-check. DOn't forget your header comments! Otherwise, it looks good!

Commit:
Reformat program to be more OOP

Add new Parser, TaskList, Ui, and Storage classes

Commit:
Divide Duke into packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add JUnit test case for TaskList

Commit:
Modify gradle to reflect correct mainClassName

Commit:
Add JavaDoc comments for Classes and Methods

Commit:
Tweak code to follow SE-EDU Java Coding Standard

Commit:
Add support for finding keywords in tasks

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/parser/Parser.java

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/task/Event.java

Commit:
Add JavaDoc for handleFind

Commit:
Fix coding style violations

Commit:
Add GUI for Duke

Shift Duke.java into duke.gui package

Commit:
Merge branch 'branch-Level-10'

Commit:
Improve code quality

Some methods are too long and some are no longer used. Proper coding
standards are also not being adhered to in some parts.

These methods may hinder code readability making it harder for other
people to understand certain parts of the code.

Let's,
* remove methods/ classes no longer in use
* refactor long methods
* adhere to coding standard

Commit:
Fix gradle issues

Change application mainClassName to duke.gui.Launcher since we are
starting Duke from Launcher now.

There is an issue getting the image files for Duke and User when
running JAR.

Change pathname so images can be retrived and JAR can be run.

Commit:
Add assertions

Index of task to get is not checked.

Invalid index results in program crashing.

Checking the index before passing it to get() allows us to pass an
error message to the user instead of crashing the pogram.

Let's assert the index to make sure it is valid before passing it to
get().

Using assert allows us to document the assumption that index is valid.

Commit:
Revert "Improve code quality"

This reverts commit d3b6fcd17bccb6d69909b004be9428ce5d35f04c.

Commit:
Improve code quality

Some methods are too long and some are no longer used. Proper coding
standards are also not being adhered to in some parts.

These methods may hinder code readability making it harder for other
people to understand certain parts of the code.

Let's,
* remove methods/ classes no longer in use
* refactor long methods
* adhere to coding standard

Removing Ui package and Ui.java as it is no longer in use after
implementing GUI.

Refactoring methods in Storage as they are too long and hard to read.

Renaming done to isDone in Task.java as it is more readable.

arcornior committed:
Merge pull request #2 from arcornior/branch-A-Assertions

Add assertions

arcornior committed:
Merge pull request #3 from arcornior/branch-A-CodeQuality

Branch a code quality

Commit:
Add support for viewing schedule
Tidy Task up by removing redundant toString call.

Commit:
Remove a few lines of blankspace to make method shorter.

Commit:
Merge branch 'branch-B-ViewSchedules'

Commit:
Add user guide

Commit:
Add Screenshot of Duke
Add Title to Duke's window

Commit:
Fix path to point to user's working directory.

Commit:
Fix link to Duke download
